community/security/src/main/java/org/neo4j/server/security/auth/AbstractUserRepository.java

```diff
@@ -89,7 +89,8 @@ public void update( User existingUser, User updatedUser )
         // Assert input is ok
         if ( !existingUser.name().equals( updatedUser.name() ) )
         {
-            throw new InvalidArgumentsException( "updated user has a different name" );
+            throw new IllegalArgumentException( "Updated user '" + updatedUser.name() + "' has a different name than " +
+                    "existing '" + existingUser.name() + "'." );
```

Comment:
This error message does not make much sense, given the context. 

Do you mean something along the following lines: "The username 'updatedUser' of the user being updated does not match the existing username 'existingUser' for the user."